src/components/Login/LoginContainer.tsx

```diff
@@ -147,14 +148,20 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
   useEffect(() => {
     const skipScanningIfParamsInDeepLink = async (): Promise<void> => {
       const { queryParams } = await Linking.parseInitialURLAsync();
-      if (queryParams?.key && queryParams?.endpoint) {
+      const queryEndpoint = queryParams?.endpoint;
+      if (queryEndpoint && !RegExp(DOMAIN_CHECK).test(queryEndpoint)) {
+        const error = new Error(`Invalid endpoint: ${queryEndpoint}`);
+        Sentry.captureException(error);
+        alert("Invalid QR code");
+        setLoginStage("SCAN");
+      } else if (queryParams?.key && queryParams?.endpoint) {
         setCodeKey(queryParams.key);
         setEndpointTemp(queryParams.endpoint);
         setLoginStage("MOBILE_NUMBER");
       }
     };
     skipScanningIfParamsInDeepLink();
-  }, []);
+  });
```

Comment:
This needs a dependency array, we only need this effect to run once

Reply:
added back!